website/home/_posts/wheres_the_graph.md

```diff
@@ -0,0 +1,148 @@
+                  context={editedReport}
+                  prompt="Summarize the HN trends in a tweet"
+                />
+              )}
+            </EditReport>
+          )}
+        </GenerateReport>
+      )}
+    </AnalyzeHNPosts>
+  )}
+</FetchHNPosts>
+```
+
+"But wait," I can hear you protesting, "where's the graph?!"
+
+It's right there. A tree is just a particular kind of graph—one where each node has exactly one parent. And that constraint buys us immense clarity without sacrificing power.
+
+## When Trees Aren't Enough, Use JavaScript
+
+Trees work beautifully for most linear workflows. But what about loops, conditionals, and cycles? Those places where traditional graphs flex their muscles?
+
+Here's where GenSX's approach really shines. Instead of inventing some custom DSL for these patterns, we use... wait for it... regular JavaScript!
+
+Need to implement an agent that loops until a condition is met? Easy:
+
+```tsx
+const AgentWorkflow = gensx.Component<{}, AgentWorkflowOutput>(
+  "AgentWorkflow",
+  <AgentStep>
+    {(result) =>
+      result.needsMoreWork ? (
+        <AgentWorkflow /> // Recursion creates a cycle
+      ) : (
+        result
+      )
+    }
+  </AgentStep>,
+);
+```
+
+Want to conditionally branch your workflow? Use a ternary operator!
+
+```tsx
+<DataFetcher url={apiUrl}>
+  {(data) =>
+    data.status === "success" ? (
+      <ProcessSuccess data={data} />
+    ) : (
+      <HandleError error={data.error} />
+    )
+  }
+</DataFetcher>
+```
+
+## The Real-World Payoff
+
+After building dozens of LLM applications, I've found this approach delivers massive benefits in day-to-day development. The workflow reads top-to-bottom like normal code, with no mental juggling required. You're not constantly context-switching between graph definition and execution flow models.
+
+Refactoring becomes vastly simpler when dependencies are explicit and components are reusable. We've all been there—trying to modify one part of a complex workflow only to discover hidden dependencies that break something seemingly unrelated. With GenSX's tree approach, those dependencies are right there in the code.
+
+Our team's velocity dramatically increased when we switched to this model. Building reusable components feels natural, not forced. You're writing code the way you'd write any other TypeScript application, using patterns you already know.
+
+## Your Graph Is Still There (Just Better)
+
+Look, I'm not anti-graph. I'm anti-unnecessary complexity. GenSX still represents your workflow as a directed graph internally. We just give you a more intuitive interface for expressing it.
+
+In tree form, data dependencies between steps are crystal clear. Each child component only sees the output of its parent through explicit props or child functions. This explicitness means no mysterious global state, no tangled webs of dependencies, and no edge spaghetti that makes you question your career choices at 2 AM.
+
+## The Tree Advantage
```

Comment:
I don't love this heading. Maybe something like "Climb a tree or get lost in a graph"